assets/training/model_management/src/validate.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Validate Import Pipeline Parameters."""
+
+import argparse
+import os
+from azure.ai.ml import MLClient
+from azureml.model.mgmt.utils.exceptions import (
+    ModelAlreadyExists,
+    swallow_all_exceptions
+)
+from azureml._common.exceptions import AzureMLException
+from azureml._common._error_definition.azureml_error import AzureMLError
+from azureml.model.mgmt.config import AppName
+from azureml.model.mgmt.utils.logging_utils import custom_dimensions, get_logger
+from azure.ai.ml.identity import AzureMLOnBehalfOfCredential
+from azure.identity import ManagedIdentityCredential
+
+logger = get_logger(__name__)
+custom_dimensions.app_name = AppName.VALIDATION_TRIGGER_IMPORT
+
+
+def _get_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--model-id", required=True)
+    return parser
+
+
+def get_mlclient(registry_name: str = None):
+    """Return ML Client."""
+    try:
+        credential = AzureMLOnBehalfOfCredential()
+        # Check if given credential can get token successfully.
+        credential.get_token("https://management.azure.com/.default")
+
+    except Exception:
+        # Fall back to ManagedIdentityCredential in case AzureMLOnBehalfOfCredential does not work
+        logger.warning("AzureMLOnBehalfOfCredential failed.")
+        try:
+            msi_client_id = os.environ.get("DEFAULT_IDENTITY_CLIENT_ID")
+            credential = ManagedIdentityCredential(client_id=msi_client_id)
+            credential.get_token("https://management.azure.com/.default")
+        except Exception:
+            logger.warning("ManagedIdentityCredential failed.")
+
+    logger.info(f"Creating MLClient with registry name {registry_name}")
+    return MLClient(credential=credential, registry_name=registry_name)
+
+
+def validate_if_model_exists(model_id):
+    """Validate if model exists in any of the registries."""
+    registries_list = ["azureml-preview", "azureml", "azureml-meta"]
```

Comment:
azureml-preview is a preview registry which most customers might not end up having access for. mlclient creation might fail for them. If this were used internally only then it make sense to have preview to but not for prod version

Reply:
@skasturi What do you think on adding azureml-preview addition. Shall we remove it.

Reply:
may be we can add the get_mlclient() in a try catch to catch exception and continue on failure.